compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -520,6 +520,7 @@ impl<'a> Parser<'a> {
         };
         let ast::FnHeader { ext, unsafety, constness, asyncness } =
             self.parse_fn_front_matter(&inherited_vis)?;
+        let span_start = self.prev_token.span;
```

Comment:
If this is a span for deleting generic parameters, then it's not recorded at the right point.
It should rather be `self.token.span` before `parse_fn_front_matter`.